helpers/chatHelper.py

```diff
@@ -266,462 +264,629 @@ async def partChannel(
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
             },
         )
-        return 404
-    except exceptions.invalidArgumentsException:
+        return ...
+
+    if osuToken.is_restricted(user_token["privileges"]):
         logger.warning(
-            "User tried to send an invalid message",
+            "User tried to send message but they are in restricted mode",
             extra={
-                "username": userToken["username"],
-                "user_id": userToken["user_id"],
-                "recipient": to,
+                "token_id": token_id,
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
             },
         )
-        return 404
-    except:
-        logger.exception("An unhandled exception occurred whle sending a chat message")
+        return ...
+
+    if await osuToken.isSilenced(user_token["token_id"]):
+        silence_time_left = await osuToken.getSilenceSecondsLeft(user_token["token_id"])
+        await osuToken.enqueue(
+            user_token["token_id"],
+            serverPackets.silenceEndTime(silence_time_left),
+        )
+
+        logger.warning(
+            "User tried to send message during silence",
+            extra={
+                "token_id": token_id,
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
+            },
+        )
+        return ...
+
+    if not message.strip():
+        logger.warning(
+            "User tried to send an empty message",
+            extra={
+                "token_id": token_id,
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
+            },
+        )
+        return ...
+
+    # optimise message
+    if len(message) > 1024:
+        message = f"{message[:1024]}... (truncated)"
+
+    # We don't want people to do shit like ()[Aika].
+    message = message.replace("()[", "[")
+
+    # There are 3 types: bot interactions, public messages and private messages
+    is_channel = send_to.startswith("#")
+    if message.startswith("!") or message.startswith("\x01"):
+        resp_code = await handle_interaction_with_bot(
+            user_token=user_token,
+            send_to=send_to,
+            message=message,
+        )
+    elif is_channel:
+        resp_code = await handle_public_message(
+            user_token=user_token,
+            send_to=send_to,
+            message=message,
+        )
+    else:
+        resp_code = await handle_private_message(
+            user_token=user_token,
+            send_to=send_to,
+            message=message,
+        )
+
+    if not osuToken.is_staff(user_token["privileges"]):
+        await osuToken.spamProtection(user_token["token_id"])
+
+    if _check_for_gamer_ids(message):
```

Comment:
lool sorry for this code, i think we can deobfuscate it to just `should_send_webhook`